### PR TITLE
options/posix: Add missing mode field to sys_openat

### PR DESCRIPTION
**Pull request by AtieP**

### CODE REVIEW

**Comment by Geertiebear** [comment]
Are these the only sysdeps that implement openat?

**Comment by AtieP** [comment]
> Are these the only sysdeps that implement openat?

Yes

**Review by ArsenArsen — COMMENTED** [review]
I think `O_TMPFILE` requires some analog code (see: https://github.com/managarm/mlibc/blob/1f09f2b08ff5f8f2cef51831d8111af4bbc9f0d0/options/posix/generic/fcntl-stubs.cpp#L70-L75)

**Comment by AtieP** [comment]
> I think `O_TMPFILE` requires some analog code (see:
> 
> https://github.com/managarm/mlibc/blob/1f09f2b08ff5f8f2cef51831d8111af4bbc9f0d0/options/posix/generic/fcntl-stubs.cpp#L70-L75
> )

Why? sys_open already featured the mode option.

File: options/posix/generic/fcntl-stubs.cpp [hunk 1]
```diff
@@ -25,12 +25,16 @@ int fcntl(int fd, int command, ...) {
 }
 
 int openat(int dirfd, const char *pathname, int flags, ...) {
+	va_list args;
+	va_start(args, flags);
+	mode_t mode = flags & O_CREAT ? va_arg(args, mode_t) : 0;
```
[the remark this answers is not in the document]
Comment (ArsenArsen) [reply options/posix/generic/fcntl-stubs.cpp:30]: +1, it'd also be nice to just not use a ternary (they're unreadable)

**Review by ArsenArsen — APPROVED** [review]

**Review by Dennisbonke — APPROVED** [review]
LGTM

**Pull request merged**